torch/utils/cpp_extension.py

```diff
@@ -262,7 +266,14 @@ def _is_binary_build() -> bool:
 
 def _accepted_compilers_for_platform() -> List[str]:
     # gnu-c++ and gnu-cc are the conda gcc compilers
-    return ['clang++', 'clang'] if IS_MACOS else ['g++', 'gcc', 'gnu-c++', 'gnu-cc', 'clang++', 'clang']
+    if IS_XPU_EXTENSION:
```

Comment:
Here will impact the logic of the CPU code base, which should use ['g++', 'gcc', 'gnu-c++', 'gnu-cc', 'clang++', 'clang'].